### ISSUE TITLE
"Path does not exist" when path definitely exists

### ISSUE THREAD

**Issue by nstr10** [docker/for-linux#674]
* [x] This is a bug report
* [ ] This is a feature request
* [x] I searched existing issues before opening this one

### Expected behavior
Using `--mount` allows mounting a directory or file from the host filesystem in a container.

### Actual behavior
```
sudo mkdir /docker
sudo chown :docker /docker
sudo chmod 777 /docker
cd /docker
docker run -d --name docker-test --mount type=bind,src="$(pwd)",dst=/test busybox
```
returns
```
docker: Error response from daemon: invalid mount config for type "bind": bind mount source path does not exist: /docker
```

### Steps to reproduce the behavior

1. Create new Ubuntu 18.04 VM in VirtualBox
2. Add official Docker repo and install latest stable
3. Question your sanity for hours when all the errors are lies

**Output of `docker version`:**

```
Client:
 Version: 18.09.6
 API Version: 1.38 (downgraded from 1.39)
 Go version: go1.10.8
 Git commit: 481bc77
 Built: Sat May 4 02:35:57 2019
 OS/Arch: linux/amd64
 Experimental: false

Server:
 Engine:
  Version: 18.06.1-ce
  Go version: go1.10.4
  Git commit: e68fc7a
  Built: Tue May 7 17:57:34 2019
  OS/Arch: linux/amd64
  Experimental: false
```

**Output of `docker info`:**

```
Can't copy from server, way too much info to type - please request specific info if needed.
```

The behavior is the same whether `/docker` is a mounted volume, a directory in the root volume, a directory under my home path, or any other kind of path I tried. Rather frustrating given this is my first install of Docker and it already seems to be trying to gaslight me.

In searching for answers, I discovered cases where people received this error for several different problems, usually for something other than missing files. This leads me to believe that the logic behind this error is very poor and it is hiding numerous other errors. As an example, this error happens in the Windows version if a user has changed their password.

**Comment by cpuguy83** [comment]
Does `/docker` exist on the machine where dockerd is running?

**Comment by nstr10** [comment]
Yes

**Comment by cpuguy83** [comment]
@nstr10 You seem to have different client and server versions... are you sure your client is talking to the server you think it is talking to?

This error *only* occurs when we stat the past in path.

**Comment by nstr10** [comment]
I'm not at all sure about that - one would assume that installing docker from the official repo would not install incompatible client and server versions.

I think this might actually be caused by AppArmor denying access but haven't had time to confirm that hypothesis. I still think the message is misleading if this is the case.

**Comment by cpuguy83** [comment]
Hmm, you may be right. Looking through the code it seems to have changed since I last had my hands on it....

https://github.com/moby/moby/blob/bd224b5fe564f343e22fa76ae8569ec949f35802/volume/mounts/linux_parser.go#L85-L88

This is not checking errors at all.

**Comment by cpuguy83** [comment]
Fix is in moby/moby#39251

**Comment by nstr10** [comment]
In case anyone comes across this before your patch filters out to releases, my case was one of AppArmor saving me from the 0-day of me.

Thank you very much for looking into this and for the patch!
